pkg/bot/slack.go

```diff
@@ -178,47 +165,54 @@ func (b *Slack) IntegrationName() config.CommPlatformIntegration {
 	return config.SlackCommPlatformIntegration
 }
 
-// NotificationsEnabled returns current notification status.
-func (b *Slack) NotificationsEnabled() bool {
-	b.notifyMutex.RLock()
-	defer b.notifyMutex.RUnlock()
-	return b.notify
+// NotificationsEnabled returns current notification status for a given channel name.
+func (b *Slack) NotificationsEnabled(channelName string) bool {
+	channel, exists := b.getChannels()[channelName]
+	if !exists {
+		return false
+	}
+
+	return channel.notify
 }
 
-// SetNotificationsEnabled sets a new notification status.
-func (b *Slack) SetNotificationsEnabled(enabled bool) error {
-	b.notifyMutex.Lock()
-	defer b.notifyMutex.Unlock()
-	b.notify = enabled
+// SetNotificationsEnabled sets a new notification status for a given channel name.
+func (b *Slack) SetNotificationsEnabled(channelName string, enabled bool) error {
+	channels := b.getChannels()
+	channel, exists := channels[channelName]
+	if !exists {
+		return execute.ErrNotificationsNotConfigured
+	}
+
+	channel.notify = enabled
+	channels[channelName] = channel
+	b.setChannels(channels)
+
 	return nil
 }
 
 func (sm *slackMessage) HandleMessage(b *Slack) error {
-	// Check if message posted in authenticated channel
-	info, err := b.Client.GetConversationInfo(sm.Event.Channel, true)
-	if err == nil {
-		if info.IsChannel || info.IsPrivate {
-			// Message posted in a channel
-			// Serve only if starts with mention
-			if !strings.HasPrefix(sm.Event.Text, "<@"+sm.BotID+">") {
-				sm.log.Debugf("Ignoring message as it doesn't contain %q prefix", sm.BotID)
-				return nil
-			}
-			// Serve only if current channel is in config
-			if b.ChannelName == info.Name {
-				sm.IsAuthChannel = true
-			}
-		}
+	// Handle message only if it starts with bot mention
+	if !strings.HasPrefix(sm.Event.Text, "<@"+sm.BotID+">") {
```

Comment:
BTW, it would be nice to have this consistent across all Bots, if `discord` already uses:
```go
r.MatchString
r.ReplaceAllString
```
here we have:
```go
strings.HasPrefix
strings.TrimPrefix
```
so, using `regex` would be more generic solution that could work in all places, so we can have some helper method around that? 🤔